content/blog/performant-elm/index.md

```diff
@@ -0,0 +1,135 @@
+---
+title: Performant Elm, Part I
+date: "2019-07-08T12:00:00.000Z"
+description: A tale of performance optimization in Elm.
+---
+
+You have written your first application in Elm. Congratulations my friend,
+the hard work has finally paid off. Now it's time to relax and enjoy the
+pure bliss that is maintaining and refactoring Elm code.
+
+But wait a second, someone tells you the app feels a bit sluggish to use.
+Mh. I thought that writing pure functions would automatically make code
+fast. Okay. Let's try to stay calm.
+
+The app looks fast at a first glance. To be fair, you're probably using a
+quad-core machine with 16 GB of RAM. Maybe you should see how the app
+behaves on a lower end device. In Chrome, you can bring up the inspector,
+then go to Performance tab and choose how much you want to throttle your
+CPU.
+
+![Chrome CPU Throttling](./chrome-throttling.png)
+
+When you click around the app now, you can clearly see it's quite sluggish.
+Mh. Time to unleash the power of the flamegraph. You can start recording a
+session, then click around the app as a normal user would. Easy peasy. When
+all is done, you should see something like this.
+
+![Chrome Performance](./chrome-performance.png)
+
+That looks quite exotic, but let's not lose hope. The yellow parts of the
+graph represent the time your browser spent executing scripts, while the
+purple parts denote the time it took to render. You can also see small red
+corners scattered through the graph: those represent when the user has
+experienced noticeable delays while using the application. You can find out
+more information about what it means
+[here](https://developers.google.com/web/fundamentals/performance/rail#goals-and-guidelines "Measure Performance with the RAIL Model").
+
+You'll also notice you can select a smaller time interval and start digging
+into more specific bits of what your application is doing at a certain
+time. For example, if you highlight the first big chunk of scripting and
+then look in the lower panel, you will find a tab named "Bottom-Up".
+
+![Bottom-up](./bottom-up.png)
```

Comment:
💭 Consider a gif here as user's new to the Performance tool might not be clear on where everything is.